File: Task-03/primes.rb
def find_primes(n)
  arr = Array.new(n -1, true)



  p = 2
  while p**2 <= n
    if arr[p-2]
      for i in (p*p ).step((n+1),p)
         arr[i-2] = false
      end
    end
    p += 1
  end

  primes = (2..n).select { |i| arr[i-2] }
  return primes
end

n=gets.chomp
n = n.to_i
prime_numbers = find_primes(n)
puts ("Prime numbers up to #{limit} are: #{prime_numbers.join(', ')}")

File: Task-03/primes.py
def find_primes(n):
    arr=[True]*(n-1)
    p=2
    prime=[]
    while p**2<=n:
        if arr[p-2]:
            for i in range(p**2,n+1,p):
                arr[i-2]=False
        p+=1
    for i in range(2,n+1):
        if arr[i-2]:
            prime.append(i)
    return prime

num = int(input("Enter a positive number: "))
print(f"Primes till {num} are{find_primes(num)}")
